### PR TITLE
Issue#4 lidl

### PR DESCRIPTION
**Pull request by cdnhaese**
Solution of the parsing 101 exercise, applied to the LIDL webshop.

### CODE REVIEW

File: exercises/parsing101/parsing.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def __init__(self, url: str, name: str =None):
         self.name = name
 
     def __str__(self):
-        return '{category}: {url}'.format(category=name, url=url)
+        return '{category}: {url}'.format(category=self.name, url=self.url)
```
Comment (sesprit) [inline exercises/parsing101/parsing.py]: Well spotted :+1:

File: solutions/parsing101-cdnhaese.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        self.name = name
+
+    def __str__(self):
+        return '{category}: {url}'.format(category=self.name, url=self.url)
+
+
+def extract_product_data(html):
+    soup = bs4.BeautifulSoup(html, 'html.parser')
+
+    nametag = str(soup.select('div > div > h1')[0])
```
Comment (sesprit) [inline solutions/parsing101-cdnhaese.py]: What if you arrive on a page where this element is not present?

File: solutions/parsing101-cdnhaese.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    pricetag_new = re.sub(r'\s+', ' ', str(soup.select('div > div > span > b')[0]))
+    # I did use string editing to remove tabs and newlines
+    pricetag_old = str(soup.select('div > div > span > em > span'))
+
+    if pricetag_old == '[]':  # If there's no promo for an article, no old_price is returned
```
Comment (sesprit) [inline solutions/parsing101-cdnhaese.py]: This is a bit weird, you make a string from a list of elements? Aren't there any better ways to check whether no promo was found? The pythonic way is to use exception handling...

**Comment by cdnhaese** [comment]
@sesprit Like in the last commit?

**Comment by sesprit** [comment]
Yep thanks!

**Pull request merged**